vpr/test/test_vpr_constraints.cpp

```diff
@@ -421,3 +423,60 @@ TEST_CASE("RegionLocked", "[vpr]") {
 
     REQUIRE(is_r2_locked == false);
 }
+
+static constexpr const char kArchFile[] = "test_read_arch_metadata.xml";
+
+// Test that place constraints are not changed during placement
+TEST_CASE("PlaceConstraintsIntegrity", "[vpr]") {
+    auto options = t_options();
+    auto arch = t_arch();
+    auto vpr_setup = t_vpr_setup();
+
+    vpr_initialize_logging();
+
+    // Command line arguments
+    const char* argv[] = {
+        "test_vpr",
+        kArchFile,
+        "wire.eblif",
+        "--fix_clusters", "wire.constraints",
+        "--place_static_move_prob", "0", "0", "0", "0", "0", "100", "0",
```

Comment:
Can you explain why these arguments are here?

Reply:
As described [here](https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/fe2bcfd16975cc1a93e057c8c9194f5d13ba4679/vpr/src/base/read_options.cpp#L1792-L1800), this option defines the probabilities to select a move type. The 6th field corresponds to the [feasible region constraints move type](https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/master/vpr/src/place/feasible_region_move_generator.cpp), where there was no check on the `is_fixed` attribute of the selected block.

Also, the RL_agent seems to be enabled by default, and it does not implement the "feasible region constraints move type" hence I disabled it.

I'll update the PR with a comment on that.